libvmi/libvmi.h

```diff
@@ -1201,6 +1201,26 @@ status_t vmi_read_va(
     size_t *bytes_read
 );
 
+/**
+ * Maps num_pages of the guest's virtual memory into host, starting at the provided vaddr.
+ * Each page will have it's own pointer in access_ptrs output array. Be aware that
+ * not all virtual pages may be allocated and in such case, the corresponding array item
+ * will be set to NULL. Remember to call munmap() on each array item afterwards.
+ *
+ * @param[in] vmi LibVMI instance
+ * @param[in] pid Pid of the virtual address space
+ * @param[in] vaddr Virtual address to read from
+ * @param[in] num_pages Number of guest pages to be mapped
+ * @param[out] access_ptrs Array of size [num_pages] containing pointers to the respective guest's pages
+ */
+status_t vmi_mmap_guest(
+    vmi_instance_t vmi,
+    vmi_pid_t pid,
+    addr_t vaddr,
```

Comment:
How about taking an `access_context_t` here instead? That would allow a caller to directly specify a physical address making the API useful in scenarios where paging has not been yet enabled (or there is no OS initialized yet).